bundles/org.openhab.core/src/test/java/org/openhab/core/cache/ByteArrayFileCacheTest.java

```diff
@@ -0,0 +1,203 @@
+    private static final File SERVICE_CACHE_FOLDER = new File(CACHE_FOLDER, SERVICE_PID);
+
+    private static final String MP3_FILE_NAME = SERVICE_CACHE_FOLDER.getAbsolutePath() + "doorbell.mp3";
+    private static final String TXT_FILE_NAME = SERVICE_CACHE_FOLDER.getAbsolutePath() + "doorbell.txt";
+
+    private static @Nullable File txt_file;
+
+    private final Logger logger = LoggerFactory.getLogger(ByteArrayFileCacheTest.class);
+
+    private @NonNullByDefault({}) ByteArrayFileCache subject;
+
+    @BeforeAll
+    public static void init() throws IOException {
+        // create temporary file
+        txt_file = createTempFile();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        subject = new ByteArrayFileCache(SERVICE_PID);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        // delete all files
+        subject.clear();
+    }
+
+    @AfterAll
+    public static void cleanUp() {
+        // delete all folders
+        SERVICE_CACHE_FOLDER.delete();
+        CACHE_FOLDER.delete();
+        USERDATA_FOLDER.delete();
+    }
+
+    @Test
+    public void testGetFileExtension() {
+        assertThat(subject.getFileExtension("/var/log/openhab2/"), is(nullValue()));
+        assertThat(subject.getFileExtension("/var/log/foo.bar/"), is(nullValue()));
+        assertThat(subject.getFileExtension("doorbell.mp3"), is(equalTo("mp3")));
+        assertThat(subject.getFileExtension("/tmp/doorbell.mp3"), is(equalTo("mp3")));
+        assertThat(subject.getFileExtension(MP3_FILE_NAME), is(equalTo("mp3")));
+        assertThat(subject.getFileExtension(TXT_FILE_NAME), is(equalTo("txt")));
+        assertThat(subject.getFileExtension("/var/log/openhab2/.."), is(""));
+        assertThat(subject.getFileExtension(".hidden"), is(equalTo("hidden")));
+        assertThat(subject.getFileExtension("C:\\Program Files (x86)\\java\\bin\\javaw.exe"), is(equalTo("exe")));
+        assertThat(subject.getFileExtension("https://www.youtube.com/watch?v=qYrpPrLY868"), is(nullValue()));
+    }
+
+    @Test
+    public void testGetUniqueFileName() {
+        String mp3UniqueFileName = subject.getUniqueFileName(MP3_FILE_NAME);
+        assertThat(mp3UniqueFileName, is(equalTo(subject.getUniqueFileName(MP3_FILE_NAME))));
+
+        String txtUniqueFileName = subject.getUniqueFileName(TXT_FILE_NAME);
+        assertThat(txtUniqueFileName, is(equalTo(subject.getUniqueFileName(TXT_FILE_NAME))));
+
+        assertThat(mp3UniqueFileName, is(not(equalTo(txtUniqueFileName))));
+    }
+
+    @Test
+    public void testGet() {
+        assertThrows(FileNotFoundException.class, () -> subject.get(TXT_FILE_NAME));
+    }
+
+    @Test
+    public void testPut() throws IOException {
+        byte[] buffer = readTempFile();
+        subject.put(TXT_FILE_NAME, buffer);
+
+        assertThat(subject.get(TXT_FILE_NAME), is(equalTo(buffer)));
+    }
+
+    @Test
+    public void testPutIfAbsent() throws IOException {
```

Comment:
Should there be a test, that `putIfAbsent ` does not overwrite an existing file?
So

- do a put with contentA
- do a putIfAbsent with contentB
- do a get and assert contentA